### ISSUE TITLE
Assert.Equivalent return incorrect result when compare decimal

### ISSUE THREAD

**Issue by msighkertsui** [xunit/xunit#2913]
Assert.Equivalent does not compare the value of two decimal value.

Sample code 1:
`Assert.Equivalent(1m, 2m) `

Expected result:
Exception throw 

Observation:
Statement run successfully

Sample code 2:
```
class TestObject {
    public decimal Value { get; set; }
}

Assert.Equivalent(new TestObject() {Value = 1m}, new TestObject() {Value = 2m}) ;
```

Expected result:
Exception throw 

Observation:
Statement run successfully

**Comment by bradwilson** [comment]
Wow, TIL that `Type.IsPrimitive` returns `false` for decimals.

**Comment by bradwilson** [comment]
Available in v2 `2.7.1-pre.15`
Available in v3 `0.1.1-pre.392`

https://xunit.net/docs/using-ci-builds
